modules/cli/src/main/scala/scala/cli/commands/util/SharedOptionsUtil.scala

```diff
@@ -23,6 +24,68 @@ import scala.util.Properties
 import scala.util.control.NonFatal
 
 object SharedOptionsUtil {
+
+  private def downloadInputs(cache: FileCache[Task]): String => Either[String, Array[Byte]] = {
+    url =>
+      val artifact = Artifact(url).withChanging(true)
+      val res = cache.logger.use {
+        try cache.withTtl(0.seconds).file(artifact).run.unsafeRun()(cache.ec)
+        catch {
+          case NonFatal(e) => throw new Exception(e)
+        }
+      }
+      res
+        .left.map(_.describe)
+        .map(f => os.read.bytes(os.Path(f, Os.pwd)))
+  }
+
+  def inputs(
```

Comment:
`SharedOptionsOps` already contains an `inputs` method, as well as a few `inputsOrExit`.
Doesn't have to be in the scope of this PR, but I think some scaladocs and/or a refactor here could prevent the future confusion on which way of constructing an `Inputs` instance is the proper one in what circumstances.

Reply:
Good point. Just pushed a few commits cleaning that up, and adding a scaladoc to the new `inputs` method.